compose/src/main/kotlin/com/yourssu/handy/compose/button/BoxButton.kt

```diff
@@ -0,0 +1,184 @@
+package com.yourssu.handy.compose.button
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.yourssu.handy.compose.HandyTheme
+import com.yourssu.handy.compose.Icon
+import com.yourssu.handy.compose.IconSize
+import com.yourssu.handy.compose.Text
+import com.yourssu.handy.compose.foundation.HandyTypography
+
+
+enum class BoxButtonType {
+    Primary,
+    Secondary,
+    Tertiary;
+}
+
+enum class BoxButtonSize {
+    XXS,
+    XS,
+    S,
+    M,
+    L,
+    XL;
+}
+
+/**
+ * BaseButton : ripple 효과가 없는 Composable 함수 입니다.
+ *
+ * BoxButton, TextButton의 베이스가 됩니다.
+ *
+ * @param onClick Button 클릭 시 실행되는 함수
+ * @param text Button 내부 text
+ * @param leftIcon Button 왼쪽에 표시되는 Icon
+ * @param rightIcon Button 오른쪽에 표시되는 Icon
+ * @param isDisabled Button 비활성화 여부
+ * @param sizeType Button 사이즈
+ * @param buttonType Button 타입 (Primary, Secondary, Tertiary)
+ * @param interactionSource Button 상호작용 소스
+ * @param horizontalPadding Button 내부 content padding
+ **/
+
+@Composable
+fun BoxButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    text: String,
+    leftIcon: ImageVector? = null,
+    rightIcon: ImageVector? = null,
+    isDisabled: Boolean = false,
+    sizeType: BoxButtonSize = BoxButtonSize.M,
+    buttonType: BoxButtonType = BoxButtonType.Primary,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    horizontalPadding: Dp = boxButtonSizeStateBySize(size = sizeType).horizontalPadding,
+) {
+    val roundingDp = boxButtonSizeStateBySize(size = sizeType).round
+    val (typo, iconSize, height) = boxButtonSizeStateBySize(size = sizeType)
+
+    BaseButton(
+        onClick = onClick,
+        colors = boxButtonColorByType(
+            type = buttonType,
+        ),
+        modifier = Modifier
+            .then(modifier)
+            .height(height),
```

Comment:
`Modifier.then`이 맨 처음에 있다면 없어도 됩니다!
```suggestion
        modifier = modifier
            .height(height),
```